src/account/sync/mod.rs

```diff
@@ -2748,13 +2754,52 @@ pub(crate) async fn repost_message(
     Ok(message)
 }
 
+fn verify_unlock_blocks(
+    transaction_payload: &TransactionPayload,
+    mut inputs: Vec<(Input, BeeAddress)>,
+) -> crate::Result<()> {
+    // Sort inputs
+    inputs.sort_by(|a, b| a.0.cmp(&b.0));
```

Comment:
That looks weird, what's happening here ?

Reply:
It's sorting the inputs by OutputId, because the need to be sorted by the protocol

Reply:
But this is currently sorting by `(Input, BeeAddress)`

Reply:
No, by `Input` because of `.0`

Reply:
Right, I'm tired, my bad. But still, you should probably sort by serialisation (pack_new) and not rely on the bytes layout, this could break on different architectures (endianness etc)